### PR TITLE
feat: use node deps dynamically

### PR DESCRIPTION
**Pull request by loks0n**
Changes to the `node-appwrite` SDK:

- Adds a `setFetch` endpoint allows a custom fetch implementation to be passed to the server API
- Made imports of `fetch`, `FormData`, `ReadableStream` and `File` dynamic. Imports are prevented in edge environments, allowing use of the native runtime implementation

### CODE REVIEW

File: templates/node/index.d.ts.twig [hunk 1]
```diff
@@ -98,6 +98,15 @@ declare module "{{ language.params.npmPackage|caseDash }}" {
...
     setSelfSigned(status?: boolean): Client;
+
+    /**
+     * Set fetch implementation.
+     *
+     * @param {any} customFetch
+     *
+     * @return {this}
+     */
+    setFetch(customFetch: any): Client;
```
Comment (ArmanNik) [inline templates/node/index.d.ts.twig:109]:
```suggestion
    setFetch(customFetch: unknown): Client;
```

Comment (ArmanNik) [reply templates/node/index.d.ts.twig:109]: We don't need `any` I believe in you

File: templates/node/lib/polyfills.js.twig [hunk 2]
```diff
@@ -0,0 +1,14 @@
+function polyfillPackage(name, package) {
+  if (typeof globalThis.EdgeRuntime !== "string" && !(name in globalThis)) {
+    try {
+      globalThis[name] = require(package)[name];
+    } catch {}
```
Comment (ArmanNik) [inline templates/node/lib/polyfills.js.twig:5]: can we catch it?

**Comment by loks0n** [comment]
Closing in favour of https://github.com/appwrite/sdk-generator/pull/799

**Pull request closed without merging**